sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -327,6 +396,17 @@ public Flux<CosmosBulkOperationResponse<TContext>> execute() {
                                 getThreadInfo());
                         }
                     });
+            })
+            .doOnCancel(() -> {
```

Comment:
a confirmation: since we have doFinally, I guess by theory this is really needed? (but it is good to have double protection)

Reply:
Good catch - meant to remove this when I refactored and made BulKExecutor disposable.